pkg/api/validation/validation.go

```diff
@@ -30,11 +30,53 @@ import (
 )
 
 // ValidateNameFunc validates that the provided name is valid for a given resource type.
-// Not all resources have the same validation rules for names.
-type ValidateNameFunc func(name string) (bool, string)
+// Not all resources have the same validation rules for names. Prefix is true if the
+// name will have a value appended to it.
+type ValidateNameFunc func(name string, prefix bool) (bool, string)
+
+// maskTrailingDash replaces the final character of a string with a subdomain safe
+// value if is a dash.
+func maskTrailingDash(name string) string {
+	if strings.HasSuffix(name, "-") {
+		return name[:len(name)-2] + "a"
+	}
+	return name
+}
+
+// ValidatePodName can be used to check whether the given pod name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+func ValidatePodName(name string, prefix bool) (bool, string) {
```

Comment:
Just today I made up a "use case" and used these in OpenShift out of spite - we are doing generation of replication controllers for each deployment so we sanity check the replication controller name validation prior to submitting it to the api, and we started generating services and I was letting the service name validation guide whether we made up a name, or used one that corresponded to the docker image's name we generated an app from.  Having them separate was truly useful.


> On Feb 2, 2015, at 6:50 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/validation/validation.go:
> 
> > +// name will have a value appended to it.
> > +type ValidateNameFunc func(name string, prefix bool) (bool, string)
> > +
> > +// maskTrailingDash replaces the final character of a string with a subdomain safe
> > +// value if is a dash.
> > +func maskTrailingDash(name string) string {
> > +	if strings.HasSuffix(name, "-") {
> > +		return name[:len(name)-2] + "a"
> > +	}
> > +	return name
> > +}
> > +
> > +// ValidatePodName can be used to check whether the given pod name is valid.
> > +// Prefix indicates this name will be used as part of generation, in which case
> > +// trailing dashes are allowed.
> > +func ValidatePodName(name string, prefix bool) (bool, string) {
> That justifies pod name - none of the others. It's minor, it's just noise without a reason.
> 
> —
> Reply to this email directly or view it on GitHub.
>